internal/incrementalcompiler-core/src/main/scala/sbt/internal/inc/MiniSetupUtil.scala

```diff
@@ -19,13 +20,20 @@ object MiniSetupUtil {
       def sameCompiler = equivComp.equiv(a.compilerVersion, b.compilerVersion)
       def sameOrder = a.order == b.order
       def sameNameHasher = a.nameHashing == b.nameHashing
+      def sameExtra = equivPairs.equiv(a.extra, b.extra)
       sameOutput &&
         sameOptions &&
         sameCompiler &&
         sameOrder && // equivOrder.equiv(a.order, b.order)
-        sameNameHasher
+        sameNameHasher &&
+        sameExtra
     }
   }
+  implicit val equivPairs: Equiv[Array[T2[String, String]]] = new Equiv[Array[T2[String, String]]] {
+    def equiv(a: Array[T2[String, String]], b: Array[T2[String, String]]): Boolean =
+      ((a.toList filter { x => !(x.get1 startsWith "info.") }) sameElements
+        (b.toList filter { x => !(x.get1 startsWith "info.") }))
+  }
```

Comment:
When the metadata is not equivalent we will invalidate the previous compilation result, but there might be a situation where we would like to track some information without invalidating the previous.